### PR TITLE
Add support for updating a password via json

### LINKED ISSUE #858: Support for JSON password resets and updates.
We need to be able to do reset requests and password updates from the front-end.

### PR DESCRIPTION
**Pull request by edpaget**
Closes #858

### CODE REVIEW

File: config/frontend_redirect.yml.hudson [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+test:
+  base: "http://localhost:2727"
+  password_reset: "/#/reset-password"
+  
+staging:
+  base: "http://demo.zooniverse.org/panoptes-front-end"
+  password_reset: "/#/reset-password"
+  
+production:
+  base: "https://panoptes.zooniverse.org/"
```
Comment (camallen) [inline config/frontend_redirect.yml.hudson]: trailing slash will result in `//`

File: config/routes.rb [hunk 2]
```diff
@@ -15,16 +15,13 @@
   end
 
   devise_for :users,
-    controllers: { omniauth_callbacks: 'omniauth_callbacks', passwords: 'passwords' },
-    skip: [ :sessions, :registrations ]
+    controllers: { omniauth_callbacks: 'omniauth_callbacks', passwords: 'passwords', registrations:  'registrations' },
```
Comment (camallen) [inline config/routes.rb]: Pretty sure we don't want to include all the devise registration controller functionality (specifically destroy) and instead just add custom routes to the `as :user` block to wire up to the registrations controller actions.

File: config/frontend_redirect.yml.hudson (hunk 1 above)
Comment (edpaget) [inline config/frontend_redirect.yml.hudson]: I don't think those will necessarily be used. I was including them as examples. We'll need to add a config file on s3 whenever we decide on deployment locationa

**Comment by brian-c** [comment]
The email I get points to `https://panoptes-staging.zooniverse.org/users/password/edit?reset_password_token=foo`, should that still be the case with this merged?

**Comment by edpaget** [comment]
Yeah it's a configuration option. You want it to go to https://demo.zooniverse.org/panoptes-front-end/#/reset-password for staging right? I'll add configuration files and restart.

**Comment by brian-c** [comment]
**http://**, no SSL on staging, right?

And https://preview.zooniverse.org/panoptes/#/ for production for right now?

**Pull request merged**